org.eclipse.sisu.plexus/src/main/java/org/codehaus/plexus/component/configurator/converters/composite/ObjectWithFieldsConverter.java

```diff
@@ -51,14 +53,31 @@ public Object fromConfiguration( final ConverterLookup lookup, final PlexusConfi
             {
                 return null; // nothing to process
             }
-            final Object bean = instantiateObject( implType );
-            if ( null == value )
+            Object bean = null;
+            // first try empty constructor with calling set/add methods or injecting fields
+            try
             {
-                processConfiguration( lookup, bean, loader, configuration, evaluator, listener );
+                bean = instantiateObject( implType );
+                if ( null == value )
+                {
+                    processConfiguration( lookup, bean, loader, configuration, evaluator, listener );
+                }
+                else
+                {
+                    new CompositeBeanHelper( lookup, loader, evaluator, listener ).setDefault( bean, value, configuration );
+                }
             }
-            else
+            catch( ComponentConfigurationException e )
             {
-                new CompositeBeanHelper( lookup, loader, evaluator, listener ).setDefault( bean, value, configuration );
+                // fallback: try with constructor taking single string
+                if ( configuration.getChildCount() == 0 && value instanceof String )
+                {
+                    bean = fromConstructorTakingString( type, (String) value, e );
+                }
+                else
+                {
+                    throw e;
+                }
```

Comment:
I would like to use https://google.github.io/guice/api-docs/3.0/javadoc/com/google/inject/spi/TypeConverter.html as well, but I don't know how to access the [Injector](https://google.github.io/guice/api-docs/3.0/javadoc/com/google/inject/Injector.html#getTypeConverterBindings()) from here (as this is independent of Plexus Container which holds the injector). @cstamas or @mcculls any idea here?

Reply:
The only option coming to my mind is passing a `PlexusContainer` to the constructor of `AbstractComponentConfigurator` (in addition to the default empty constructor) and then pass that instance in calls to `fromConfiguration` of `ConfigurationConverter`. Additionally the `PlexusContainer` would need to be casted to `DefaultPlexusContainer` to somehow extract the Guice `Injector` from there in `ObjectWithFieldsConverter`.